<a href="https://colab.research.google.com/github/wahyunh10/Superstore-Project/blob/main/Superstore_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**US Superstore Data Exploration**

This dataset contains transactions from the sale has recorded in the US Superstore data from 2014-2018.

This Dataset Can Be Viewed At :[Dataset](https://www.kaggle.com/juhi1994/superstore).

This dataset has 21 columns and 9994 rows.

From 21 columns you can know the Customer's Name, Order Date, Product Name, Category, Sub-Category, Selling Price, Discounts, Profits, and Other Columns.

In [ ]:
pip install --upgrade xlrd

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 96 kB 3.1 MB/s 
  Attempting uninstall: xlrd
    Found existing installation: xlrd 1.1.0
    Uninstalling xlrd-1.1.0:
      Successfully uninstalled xlrd-1.1.0


In [ ]:
pip install geopandas

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 1.0 MB 4.1 MB/s 
     |████████████████████████████████| 16.7 MB 7.9 MB/s 
     |████████████████████████████████| 6.3 MB 11.1 MB/s 


In [ ]:
pip install adjustText

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Created wheel for adjustText: filename=adjustText-0.7.3-py3-none-any.whl size=7097 sha256=6ad0517b4184e6f29ffd43b9ed4c7563e6cfb93763762ef37fbe9fe342a30902
  Stored in directory: /root/.cache/pip/wheels/2f/98/32/afbf902d8f040fadfdf0a44357e4ab750afe165d873bf5893d
Successfully built adjustText


In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.set
import geopandas
import warnings
warnings.filterwarnings('ignore')
from adjustText import adjust_text

In [17]:
df = pd.read_excel('US Superstore data.xls')
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
df[['Sales', 'Quantity', 'Discount', 'Profit']].describe()

NameError: ignored

It can be seen that there is a minus on profit column on the transaction data with row ID 4.

After checking with the statistics description, it turns out that the smallest value of profit is -6599.978000.

This indicates the transactions disadvantageous from all transactions recorded in the US Superstore data.

In [ ]:
profitDF = df['Profit'] > 0
profitDF = df[profitDF]
profitPercentage = profitDF.shape[0]/df.shape[0]*100
print('Profit Transactions :', profitDF.shape[0], 'Transactions Or', "{:.2f}".format(profitPercentage), '%')

lossesDF = df['Profit'] < 0
lossesDF = df[lossesDF]
lossesPercentage = lossesDF.shape[0]/df.shape[0]*100
print('Losses Transactions :', lossesDF.shape[0], 'Transactions Or', "{:.2f}".format(lossesPercentage), '%')

zeroDF = df['Profit'] == 0
zeroDF = df[zeroDF]
zeroPercentage = zeroDF.shape[0]/df.shape[0]*100
print('Zero Transactions   :', ' ',zeroDF.shape[0], 'Transactions Or ', "{:.2f}".format(zeroPercentage), '%')

Profit Transactions : 8058 Transactions Or 80.63 %
Losses Transactions : 1871 Transactions Or 18.72 %
Zero Transactions   :   65 Transactions Or  0.65 %


After a deeper check it turned out that there were 1871 losses transactions or abaout 18% of all transactions.

Meanwhile, 8058 profitable transactions and there are 65 transactions made 0 profit.

So it needs more deep investigation losses and anythings that can increase profits.


# **Product Analyst**

In [ ]:
product = df.groupby(['Product Name', 'Category', 'Sub-Category']).agg({'Sales' : 'mean', 'Quantity' : 'sum', 'Profit' : 'sum'})

In [ ]:
profitProduct = product['Profit'] > 0
profitProduct = product[profitProduct]
profitProduct = profitProduct.sort_values('Profit', ascending = False)
profitProduct

Sales  \
Product Name                                       Category        Sub-Category                
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers       12319.96480   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        2745.33840   
Hewlett Packard LaserJet 3310 Copier               Technology      Copiers        2354.96075   
Canon PC1060 Personal Laser Copier                 Technology      Copiers        2904.95850   
HP Designjet T520 Inkjet Large Format Printer -... Technology      Machines       6124.96500   
...                                                                                      ...   
Sony 8GB Class 10 Micro SDHC R40 Memory Card       Technology      Accessories      27.19200   
Letter Slitter                                     Office Supplies Supplies          5.04000   
Bagged Rubber Bands                                Office Supplies Fasteners         5.14080   
HON 5400 Series Task Chairs for Big and Tall       Furniture       Chairs         2733.82200   
Avery Trapezoid Extra Heavy Duty 4" Binders        Office Supplies Binders          78.63750   

                                                                                 Quantity  \
Product Name                                       Category        Sub-Category             
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers             20   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders             31   
Hewlett Packard LaserJet 3310 Copier               Technology      Copiers             38   
Canon PC1060 Personal Laser Copier                 Technology      Copiers             19   
HP Designjet T520 Inkjet Large Format Printer -... Technology      Machines            12   
...                                                                                   ...   
Sony 8GB Class 10 Micro SDHC R40 Memory Card       Technology      Accessories          3   
Letter Slitter                                     Office Supplies Supplies             4   
Bagged Rubber Bands                                Office Supplies Fasteners           21   
HON 5400 Series Task Chairs for Big and Tall       Furniture       Chairs              39   
Avery Trapezoid Extra Heavy Duty 4" Binders        Office Supplies Binders             15   

                                                                                       Profit  
Product Name                                       Category        Sub-Category                
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers       2.519993e+04  
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders       7.753039e+03  
Hewlett Packard LaserJet 3310 Copier               Technology      Copiers       6.983884e+03  
Canon PC1060 Personal Laser Copier                 Technology      Copiers       4.570935e+03  
HP Designjet T520 Inkjet Large Format Printer -... Technology      Machines      4.094977e+03  
...                                                                                       ...  
Sony 8GB Class 10 Micro SDHC R40 Memory Card       Technology      Accessories   3.399000e-01  
Letter Slitter                                     Office Supplies Supplies      3.024000e-01  
Bagged Rubber Bands                                Office Supplies Fasteners     3.024000e-01  
HON 5400 Series Task Chairs for Big and Tall       Furniture       Chairs        3.979039e-13  
Avery Trapezoid Extra Heavy Duty 4" Binders        Office Supplies Binders       2.131628e-14  

[1547 rows x 3 columns]

In [ ]:
print('Total Profit From Product :', profitProduct['Profit'].sum())

Total Profit From Product : 363465.4024999999


In the profit product table, shows all the products name that were sold from the US Superstore data.

There are 1547 different products sold and generated profit about 363465.40.

In [ ]:
bestProduct = profitProduct.head(10)
bestProduct

Sales  \
Product Name                                       Category        Sub-Category                 
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers       12319.964800   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        2745.338400   
Hewlett Packard LaserJet 3310 Copier               Technology      Copiers        2354.960750   
Canon PC1060 Personal Laser Copier                 Technology      Copiers        2904.958500   
HP Designjet T520 Inkjet Large Format Printer -... Technology      Machines       6124.965000   
Ativa V4110MDD Micro-Cut Shredder                  Technology      Machines       3849.945000   
3D Systems Cube Printer, 2nd Generation, Magenta   Technology      Machines       7149.945000   
Plantronics Savi W720 Multi-Device Wireless Hea... Technology      Accessories    1338.184286   
Ibico EPK-21 Electric Binding System               Office Supplies Binders        5291.972000   
Zebra ZM400 Thermal Label Printer                  Technology      Machines       3482.850000   

                                                                                 Quantity  \
Product Name                                       Category        Sub-Category             
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers             20   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders             31   
Hewlett Packard LaserJet 3310 Copier               Technology      Copiers             38   
Canon PC1060 Personal Laser Copier                 Technology      Copiers             19   
HP Designjet T520 Inkjet Large Format Printer -... Technology      Machines            12   
Ativa V4110MDD Micro-Cut Shredder                  Technology      Machines            11   
3D Systems Cube Printer, 2nd Generation, Magenta   Technology      Machines            11   
Plantronics Savi W720 Multi-Device Wireless Hea... Technology      Accessories         24   
Ibico EPK-21 Electric Binding System               Office Supplies Binders             13   
Zebra ZM400 Thermal Label Printer                  Technology      Machines             6   

                                                                                     Profit  
Product Name                                       Category        Sub-Category              
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers       25199.9280  
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        7753.0390  
Hewlett Packard LaserJet 3310 Copier               Technology      Copiers        6983.8836  
Canon PC1060 Personal Laser Copier                 Technology      Copiers        4570.9347  
HP Designjet T520 Inkjet Large Format Printer -... Technology      Machines       4094.9766  
Ativa V4110MDD Micro-Cut Shredder                  Technology      Machines       3772.9461  
3D Systems Cube Printer, 2nd Generation, Magenta   Technology      Machines       3717.9714  
Plantronics Savi W720 Multi-Device Wireless Hea... Technology      Accessories    3696.2820  
Ibico EPK-21 Electric Binding System               Office Supplies Binders        3345.2823  
Zebra ZM400 Thermal Label Printer                  Technology      Machines       3343.5360


From 1547 profitable products in the profit product table, show 10 products that generated the biggest profits.

Canon ImageCLASS 2200 advanced copier is the product that producs the biggest profit, with 25199.92 of 20 products sold.

Followed by followers PB500 electric punch plastic comb binding machine with manual bind with 7753.03, Hewlwtt Packcard Laser Jet 3310 Copier with 6983.88, and 7 other products.

In [ ]:
lossProduct = product['Profit'] < 0
lossProduct = product[lossProduct]
lossProduct = lossProduct.sort_values('Profit', ascending = True)
lossProduct

In [ ]:
print('Total Losses From Product :', lossProduct['Profit'].sum())

In the loss products table above, there are 301 lines, which indicate products with minus profit with -77068.38


In [ ]:
worstProduct = lossProduct.head(10)
worstProduct